src/components/button/_macro.spec.js

```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
+                expect($('.ons-btn').hasClass('ons-js-submit-btn')).toBe(true);
+            });
         });
+        describe('WHEN: variants contains print', () => {
+            it('THEN: renders provided variant style classes', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        variants: 'print',
+                    }),
+                );
+
+                expect($('.ons-btn').hasClass('ons-btn--print')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-u-d-no')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-print-btn')).toBe(true);
+            });
 
-        it('defaults to being a "submit" button when `type` is not provided', () => {
-            const $ = cheerio.load(renderComponent('button'));
-
-            expect($('button').attr('type')).toBe('submit');
-        });
+            it('THEN: renders print icon when variants contains print', () => {
+                const faker = templateFaker();
+                const iconsSpy = faker.spy('icon');
 
-        it('defaults to being a "button" when `type` is not provided and `variants` contains "print"', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
+                faker.renderComponent('button', {
+                    url: 'http://example.com',
                     variants: 'print',
-                }),
-            );
+                });
 
-            expect($('button').attr('type')).toBe('button');
+                expect(iconsSpy.occurrences[0].iconType).toBe('print');
+            });
         });
+    });
+    describe('GIVEN: Params: type', () => {
+        describe('WHEN: type parameter is provided', () => {
+            it('THEN: renders the provided type attribute', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        type: 'special-type',
+                    }),
+                );
+
+                expect($('button').attr('type')).toBe('special-type');
+            });
 
-        it('has the provided `value` attribute', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    value: 'special-value',
-                }),
-            );
+            it('THEN: renders the provided type attribute if variant contains print', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        type: 'special-type',
+                        variants: 'print',
+                    }),
+                );
 
-            expect($('button').attr('value')).toBe('special-value');
+                expect($('button').attr('type')).toBe('special-type');
+            });
         });
+        describe('WHEN: type parameter is not provided', () => {
+            it('THEN: defaults to being a submit button when type is not provided', () => {
```

Comment:
```suggestion
            it('THEN: defaults to being a submit button', () => {
```
This is already in the when